Add WKUIDelegate SPI for ending XR session

achan00 opened the pull request:
#### 13389a7c3f721abba70e9b339f96dc3df64ca882
<pre>
Add WKUIDelegate SPI for ending XR session
<a href="https://bugs.webkit.org/show_bug.cgi?id=242643">https://bugs.webkit.org/show_bug.cgi?id=242643</a>

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::endXRSession):
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::setDelegate):
(WebKit::UIDelegate::UIClient::endXRSession):

Canonical link: <a href="https://commits.webkit.org/252404@main">https://commits.webkit.org/252404@main</a>
</pre>

hortont424 commented on Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
```diff
@@ -194,6 +194,7 @@ struct UIEdgeInsets;
 - (void)_webView:(WKWebView *)webView requestPermissionForXRSessionOrigin:(NSString *)originString mode:(_WKXRSessionMode)mode grantedFeatures:(_WKXRSessionFeatureFlags)grantedFeatures consentRequiredFeatures:(_WKXRSessionFeatureFlags)consentRequiredFeatures consentOptionalFeatures:(_WKXRSessionFeatureFlags)consentOptionalFeatures completionHandler:(void (^)(_WKXRSessionFeatureFlags))completionHa ...
 - (void)_webView:(WKWebView *)webView startXRSessionWithCompletionHandler:(void (^)(id))completionHandler WK_API_AVAILABLE(macos(12.0), ios(15.0));
 - (void)_webView:(WKWebView *)webView requestNotificationPermissionForSecurityOrigin:(WKSecurityOrigin *)securityOrigin decisionHandler:(void (^)(BOOL))decisionHandler WK_API_AVAILABLE(macos(10.13.4), ios(16.0));
+- (void)_webViewEndXRSession:(WKWebView *)webView WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
```
Something strikes me as odd about the delegate name, but I can't put it into words. Is this "did end"?

Anyway, this is fine.

achan00 committed:
Add WKUIDelegate SPI for ending XR session
https://bugs.webkit.org/show_bug.cgi?id=242643

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/API/APIUIClient.h:
(API::UIClient::endXRSession):
* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.h:
* Source/WebKit/UIProcess/Cocoa/UIDelegate.mm:
(WebKit::UIDelegate::setDelegate):
(WebKit::UIDelegate::UIClient::endXRSession):

Canonical link: https://commits.webkit.org/252404@main
